### PR TITLE
[WIP] Feat/non anonymous reflections

### PR DESCRIPTION
**Pull request by good-idea**
Addresses #3710 

Allows users to toggle the anonymity of their reflections.

To do:

- [x] Update graphql types and mutations
- [x] update client queries, fragments, mutations
- [x] add toggle when creating reflection
- [x] preliminary styles and add user avatar
- [x] get subscriptions working on Group phase
- [ ] TBD review for vote, discuss, summary phases
- [ ] Update `isAnonymous` input handling in GroupingKanbanColumn.tsx (Marked with TODO)

### UX / Design review
- [ ] UX/BUG (group phase) clicking toggle on a "stacked" card opens the modal
- [ ] UX/BUG (reflect phase) initial click on anonymity button places card into a group

### CODE REVIEW

File: packages/server/database/types/Reflection.ts [hunk 1]
```diff
@@ -56,7 +59,9 @@ export default class Reflection {
     this.plaintextContent = plaintextContent || extractTextFromDraftString(content)
     this.entities = entities
     this.isActive = true
+    this.isAnonymous = true
```
Comment (mattkrick) [inline packages/server/database/types/Reflection.ts:62]: stylistically, i'd probably use the nullish coalescing `this.isAnonymous = isAnonymous ?? true`, but it all does the same thing

File: packages/client/components/GroupingKanbanColumn.tsx [hunk 2]
```diff
@@ -88,7 +88,9 @@ const GroupingKanbanColumn = (props: Props) => {
       content: undefined,
       meetingId,
       retroPhaseItemId: promptId,
-      sortOrder: getNextSortOrder(reflectionGroups)
+      sortOrder: getNextSortOrder(reflectionGroups),
+      // TODO
```
Comment (mattkrick) [inline packages/client/components/GroupingKanbanColumn.tsx:92]: looks like a good stopping point

File: packages/client/components/RetroReflectPhase/PhaseItemEditor.tsx [hunk 3]
```diff
@@ -46,12 +48,14 @@ const PhaseItemEditor = (props: Props) => {
...
     setCardsInFlight,
-    dataCy
+    dataCy,
+    currentViewer
   } = props
   const atmosphere = useAtmosphere()
   const {onCompleted, onError, submitMutation} = useMutationProps()
   const [editorState, setEditorState] = useState(EditorState.createEmpty)
   const [isEditing, setIsEditing] = useState(false)
+  const [isAnonymous, setIsAnonymous] = useState(true)
```
Comment (mattkrick) [inline packages/client/components/RetroReflectPhase/PhaseItemEditor.tsx:58]: from a design perspective, why have a discrete state for each editor vs. a single `isAnonymous` state for the whole phase? no right or wrong, just curious for your thoughts

**Comment by good-idea** [comment]
> what other changes would you make given enough time?

I think most of this would be directed by what we decided as the best UX. Some things I was looking at:

- The placement of the toggle button / avatar, and placeholder text. How I have it now is a little bulky, as every card you add comes with another button and decision to make. Putting the `isAnonymous` "higher up" (such as in the bottom bar) would probably be a better solution. But, it might mean that the toggle would mean that *all* of a user's reflections have the same setting - being able to control this per-reflection is nice.
- Following up on the above, a nice-to-have would be keeping the `isAnonymous` state consistent when adding multiple new reflections. Right now a new one always defaults to anonymous. If i just created a public reflection, the next one should be set to public as well.
- I think the introduction of the "Incognito" icon for anonymous comments might make users feel like they ought to make all of their reflections public. I think the default anonymity is nice to have and encourages more honest reflections - so, the UX ought to be adjusted to make users still feel good about keeping anonymous.

> from a design perspective, why have a discrete state for each editor vs. a single isAnonymous state for the whole phase? no right or wrong, just curious for your thoughts

> what are the pros/cons of having the toggle always being in the footer (as it is today) vs. being off to the side like the input for comments in the discuss phase?

(see above)

> what changes would you make to the codebase or patterns overall?

The structure of the components that bring this all together for the Reflect & Grouping phases felt a little convoluted - lots of props (mostly functions) are passed through in a way that doesn't leave the component too readable. I was able to figure it out, but ideally I'd like a developer to be able to look at one of the components and quickly get a sense of how it is used and operates.

For instance, `<ReflectionEditorWrapper />` is used within both `<ReflectionCard />` and `<PhaseItemEditor />`. These parent components have some duplicate logic that might be better handled within `<ReflectionCard />` itself - things such as submitting new reflections and updates. But, I know there are lots of things to consider here -- cards being dragged, handling paste input, etc -- and I don't have a full sense of how it all fits together, so it would take some work & planning. It would be great if the API for `<ReflectionCard />` was as simple as:

```js
<ReflectionCard reflection={reflection} />
```

Also, I think introducing context for some of this functionality might make things easier - even simple ones like `useMeeting()` and `useCurrentViewer()` would mean less prop drilling and, if the data requirements are a little more standardized, might also mean reducing the need to pull in data using Relay fragments.

One of the biggest benefits for breaking these things up a little bit would be testability - right now, writing tests for these components would be difficult as it would require mocking out quite a lot of data & functions.

> how was it working with relay? any gotchas?

No real gotchas other than the things we went over when chatting. But, I definitely need to brush up on Relay.

**Comment by mattkrick** [comment]
Closing until we can find a design to work with the minimal UI

**Pull request closed without merging**